addons/mail/static/src/discuss/core/common/thread_model_patch.js

```diff
@@ -68,6 +68,12 @@ const threadPatch = {
                 return this.computeCorrespondent();
             },
         });
+        this.correspondentCountry = fields.One("res.country", {
+            /** @this {import("models").Thread} */
+            compute() {
+                return this.correspondent?.persona?.country_id ?? this.country_id;
```

Comment:
For later: `country_id` field on channel should actually be moved to livechat, and even more so could be completely removed (we should always have the country on partner or guest instead)